### DISCUSSION TITLE
How do I add a default value for a one-to-one relation field?

### DISCUSSION THREAD

**Discussion by puchm** [prisma/prisma#9700] [Q&A]
Let's assume I have a model `User` and a model `Settings` which are related in a one-to-one relation:
```prisma
model User {
    id String @db.Uuid @default(uuid()) @id
    settings Settings @relation(fields: [settingsId], references: [id])
    settingsId String @db.Uuid
}

model Settings {
    id String @db.Uuid @default(uuid()) @id
    user User?
    ENABLE_X Boolean
}
```

Is it possible to create a new `Settings` record whenever a user is created? I would like to specify the values of said record in the `@default()` attribute if that is possible somehow and not specify them in the `Settings` model. Note that this is just a simplification of my situatuion - I am not able to merge `Settings` into `User`. I can also not make `settingsId` point to a default settings record - it has to be a new one.

Something I have thought about was creating a middleware. That would mean that I would have to make the `settings` in `User` optional but other than that it should be pretty straight forward.

What do you think?

**Comment by ryands17** [comment]
@puchm 👋 

There's not default for relations, but there are a couple ways to achieve this:

1. Using middleware as you mentioned, but you would then need to keep the relation optional
2. Adding a `create` for settings on every user's create. This can be done by a function like this for e.g.

```ts
await prisma.user.create({
  data: {
    ...createSettings(),
  }
})

function createSettings() {
  return { settings: { create: { ENABLE_X: false } } }
}
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
